ocaml-lsp-server/src/document.ml

```diff
@@ -187,13 +191,14 @@ let get_impl_intf_counterparts uri =
            ~message:"provided file URI (param) doesn't follow URI spec" ())
   in
   let fname = Filename.basename fpath in
-  let ml, mli, re, rei, mll, mly = ("ml", "mli", "re", "rei", "mll", "mly") in
+  let ml, mli, eliom, eliomi, re, rei, mll, mly =
+    ("ml", "mli", "eliom", "eliomi", "re", "rei", "mll", "mly") in
   let exts_to_switch_to =
     match Syntax.of_fname fname with
     | Ocaml -> (
       match Kind.of_fname fname with
-      | Intf -> [ ml; mly; mll; re ]
-      | Impl -> [ mli; mly; mll; rei ] )
+      | Intf -> [ ml; mly; mll; eliom; re ]
+      | Impl -> [ mli; mly; mll; eliomi; rei ] )
```

Comment:
Pretty sure this is wrong and it's not possible to have a `foo.ml` and `foo.eliomi` pair. It should only be possible to switch from `eliom` to `eliomi` and vice versa. @drup can you confirm?

Reply:
In theory, it's fine to have any kind of paring. I'm not sure any current build system supports it though.

Reply:
So the current makefile/ocamlbuild build systems only support `eliom`/`eliomi`. I suppose we should reflect this until eliom support is added to dune and this can be revised.